Updating webhook docs (#12764)
pulumi/docs: content/docs/esc/webhooks.md

---
title_tag: "ESC Webhooks"

> meta_desc: ESC Webhooks allow you to notify external services of events happening within your Pulumi organization. Learn how to create and manage webhooks here.

Reviewer: ```suggestion
meta_desc: ESC Webhooks allow you to notify external services of events happening within your Pulumi environments. Learn how to create and manage webhooks here.
```

title: "Webhooks"
h1: ESC Webhooks
meta_image: /images/docs/meta-images/docs-meta.png
menu:
  pulumiesc:
    weight: 9
aliases:
- /docs/intro/environment/webhooks/
- /docs/intro/environments/webhooks/
- /docs/intro/esc/webhooks/
- /docs/environment/webhooks/
- /docs/environments/webhooks/

> - /docs/esc/webhooks/

Author: Ah, ack, I thought these were for people to find it easier

---

{{% notes "info" %}}
ESC Webhooks is a feature available on the Pulumi Team, Enterprise and Business Critical editions.
To try it out, start a [trial](https://app.pulumi.com/site/trial) now.
{{% /notes %}}

> ESC Webhooks allow you to notify external services of events
> happening within your Pulumi organization. For example,

Reviewer: ```suggestion
ESC Webhooks allow you to notify external services of events
happening within your ESC environments. For example,
```

you can trigger a notification whenever a new revision of an environment is created.

> Whenever an event occurs, Pulumi will send an HTTP `POST` request to

Reviewer: ```suggestion
When an event occurs, Pulumi will notify the registered webhook listeners via a HTTP `POST` request with metadata about the event
```

> Whenever an event occurs, Pulumi will send an HTTP `POST` request to
> all registered webhooks. The webhook can then be used to emit a

Reviewer: ```suggestion
When an event occurs, Pulumi will notify the registered webhook listeners via a HTTP `POST` request with metadata about the event. The webhook can then be used to emit a
```

Author: This one is copy-pasted from original page. Should I change both or leave it be? I think we want to stay in sync

Reviewer: either way works

notification, start running integration tests, or even update Pulumi stacks.

> Webhooks can be used for pretty much anything you want, and are the foundation

Reviewer: ```suggestion
There are large number of real life applications for webhooks including serving as the foundation```

Author: Same here, copypasta from original page

of most _ChatOps_ workflows.

## Overview

ESC Webhooks can be attached to either a environment or an organization. Environment webhooks
will be notified of events specific to the environment. Organization
webhooks will be notified for events happening within each of the organization's
environments.

> The Webhooks page for organization is under Organization Settings tab, and for Environments is a tab on each Environment page

Reviewer: ```suggestion
Organization webhooks can be managed on the Organization Settings page. Environment webhooks can be managed from the webhooks tab on each Environment's detail page.
```

| `environment_revision_tag_updated`	| `environment_revision_tag`		  | Organization or Environment| When a revision tag is updated.              	  |
| `environment_tag_created`				    | `environment_tag`					      | Organization or Environment| When a new environment tag is created.           |
| `environment_tag_deleted`				    | `environment_tag`					      | Organization or Environment| When an environment tag is deleted.              |
| `environment_tag_updated`				    | `environment_tag`					      | Organization or Environment| When an environment tag is updated.              |
| `imported_environment_changed`		  | `imported_environment_changed`	| Organization or Environment| When an imported environment was changed.        |

There is also an `environments` filter group that lets you easily subscribe to all environment events.

| Group			    | Event Kinds Included 																									                                              |
|---------------|---------------------------------------------------------------------------------------------------------------------|

> |`environments`	|`environment`, `environment_revision`, `environment_revision_tag`, `environment_tag`, `imported_environment_changed`	|

Reviewer: I think you might have used tabs here instead of spaces

Author: Yep, I did😩 
I'll redo into spaces
